### PR TITLE
[Unit-7]Basic chapter of video and video processing

### PR DESCRIPTION
**Pull request by DiwakarBasnet**
Added initial topics for Introduction and Video Processing Basics. Related to issue: #19

### CODE REVIEW

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+## What is a video?
+A binary, two-dimensional (2D) representation of visual data is called an image. In essence, a video is a multimedia format that sequentially displays these frames, or images.
+Technically speaking, the frames are separate pictures. As a result, storing and playing these frames sequentially at a conventional speed results in the creation of a video, thus giving the illusion of motion (just like a flipbook).
+It is a popular and widely used medium for communicating information, entertainment, and conversation. Videos, photos, are obtained via image-acquisition equipment such as video cameras, smartphones, and so on.
+
+### Aspects of a video
+1. **Resolution:** <br />
```
Comment (mmhamdy) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:8]: The line break tag isn't really necessary.

File: chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+There are two main approaches for extracting tokens from video or embedding a video clip:
+
+### Uniform Frame Sampling:
+It is a straightforward method of tokenizing the input video in which we uniformly sample 'nt' frames from the input video clip, embed each 2D frame independently using the same method as used in image processing and concatenate all these tokens together.
+If 'nh.nw' non-overlapping image patches are extracted from each frame, then a total of 'nt.nh.nw' tokens will be forwarded through the transformer encoder. Uniform frame sampling is a tokenization scheme in which we sample frames from the video clip and perform simple ViT tokenization.
+
+### Tubelet Embedding:
+This method is an extension of the vision transformer's image embedding to 3D and corresponds to a 3D convolution. It is an alternative method in which non-overlapping, spatio-temporal “tubes” from input volume are extracted and linearly projected.
```
Comment (mmhamdy) [inline chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx:25]:
```suggestion
This method is an extension of the vision transformer's image embedding to 3D and corresponds to a 3D convolution. It is an alternative method in which non-overlapping, spatiotemporal “tubes” from input volume are extracted and linearly projected.

```

File: chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+First, we extract tubes from the video, these tubes contain patches of the frame and the temporal information as well. The tubes are then flattened to build video tokens. Intuitively, this method fuses spatio-temporal information during tokenization, in contrast to “Uniform frame sampling” where temporal information from different frames is fused by the transformer.
```
Comment (mmhamdy) [inline chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx:26]:
```suggestion
First, we extract tubes from the video, these tubes contain patches of the frame and the temporal information as well. The tubes are then flattened to build video tokens. Intuitively, this method fuses spatiotemporal information during tokenization, in contrast to “Uniform frame sampling” where temporal information from different frames is fused by the transformer.

```

**Review by mmhamdy — COMMENTED** [review]
Looks good, and nicely written 👍👍

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+## What is a video?
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:2]:
```suggestion
## What is a Video?

```

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+## What is a video?
+A binary, two-dimensional (2D) representation of visual data is called an image. In essence, a video is a multimedia format that sequentially displays these frames, or images.
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:3]:
```suggestion
An image is a  binary, two-dimensional (2D) representation of visual data. A video is a multimedia format that sequentially displays these frames or images.

```
Active voice reads easier

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+## What is a video?
+A binary, two-dimensional (2D) representation of visual data is called an image. In essence, a video is a multimedia format that sequentially displays these frames, or images.
+Technically speaking, the frames are separate pictures. As a result, storing and playing these frames sequentially at a conventional speed results in the creation of a video, thus giving the illusion of motion (just like a flipbook).
+It is a popular and widely used medium for communicating information, entertainment, and conversation. Videos, photos, are obtained via image-acquisition equipment such as video cameras, smartphones, and so on.
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:5]:
```suggestion
It is a popular and widely used medium for communicating information, entertainment, and conversation. Videos and photos are obtained via image-acquisition equipment such as video cameras, smartphones, and so on.

```

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+## What is a video?
+A binary, two-dimensional (2D) representation of visual data is called an image. In essence, a video is a multimedia format that sequentially displays these frames, or images.
+Technically speaking, the frames are separate pictures. As a result, storing and playing these frames sequentially at a conventional speed results in the creation of a video, thus giving the illusion of motion (just like a flipbook).
+It is a popular and widely used medium for communicating information, entertainment, and conversation. Videos, photos, are obtained via image-acquisition equipment such as video cameras, smartphones, and so on.
+
+### Aspects of a video
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:7]:
```suggestion
### Aspects of a Video

```
maybe attributes?

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Introduction
+## What is a video?
+A binary, two-dimensional (2D) representation of visual data is called an image. In essence, a video is a multimedia format that sequentially displays these frames, or images.
+Technically speaking, the frames are separate pictures. As a result, storing and playing these frames sequentially at a conventional speed results in the creation of a video, thus giving the illusion of motion (just like a flipbook).
+It is a popular and widely used medium for communicating information, entertainment, and conversation. Videos, photos, are obtained via image-acquisition equipment such as video cameras, smartphones, and so on.
+
+### Aspects of a video
+1. **Resolution:**
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:8]:
```suggestion
- **Resolution**

```

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+Technically speaking, the frames are separate pictures. As a result, storing and playing these frames sequentially at a conventional speed results in the creation of a video, thus giving the illusion of motion (just like a flipbook).
+It is a popular and widely used medium for communicating information, entertainment, and conversation. Videos, photos, are obtained via image-acquisition equipment such as video cameras, smartphones, and so on.
+
+### Aspects of a video
+1. **Resolution:**
+The resolution of a video refers to the number of pixels in each frame or we can also refer to it as the size of each frame in the video. It doesn't need to be a standard size, but there are common sizes for video. Common video resolutions include HD (1280x720 pixels), Full HD (1920x1080 pixels), Ultra HD or 4K (3840x2160 pixels), and so on.
+When a video is said to have a resolution of 1920x1080 pixels, it essentially means the video has a width of 1920 pixels and a height of 1080 pixels.
+Higher resolution videos have more detail but also require more storage space and processing power.
+
+2. **Frame Rate:**
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:13]:
```suggestion
- **Frame Rate**

```

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+A video is composed of multiple separate frames, or images. In order to give the impression of motion, these frames are displayed quickly one after the other.
+ The number of frames displayed per second is called the "frame rate." Common frame rates include 24, 30, and 60 frames per second (fps) or hertz (general unit for frequency). Higher frame rates result in smoother motion.
+
+3. **Bitrate:**
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:17]:
```suggestion
- **Bitrate**

```

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+The quantity of data needed to describe audio and video is referred to as bitrate. Better quality is achieved at higher bitrates, but streaming requires more storage and bandwidth.
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:18]:
```suggestion
The quantity of data needed to describe audio and video is called bitrate. Better quality is achieved at higher bitrates, but streaming requires more storage and bandwidth.

```

File: chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+Bitrates for videos are commonly expressed in megabytes per second (mbps) or kilobytes per second (kbps).
+
+4. **Codecs:**
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/introduction-to-video.mdx:21]:
```suggestion
- **Codecs**

```

File: chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx [hunk 13]
```diff
@@ -0,0 +1,34 @@
+# Video Processing Basics
+With the rise and advancements of transformers, vision transformers have become essential tools for various computer vision tasks. Vision transformers are great at computer vision tasks involving both images and videos.
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx:2]:
```suggestion
With the rise of transformers, vision transformers have become essential tools for various computer vision tasks. Vision transformers are great at computer vision tasks involving both images and videos.

```

File: chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+Suppose we have a 224x224 image, split the image into 16x16 patches where each patch consists of 14x14 pixels. This patch-based processing not only reduces computation but it also allows the model to capture local features in the image effectively.
+Each patch is then fed through a series of self-attention layers and feed-forward neural networks to extract semantic information. Thanks to this hierarchical processing technique, vision transformers are able to capture both high-level and low-level features in the image.
+
+Since vision transformers process patches individually and transformers by default don't have any mechanism to track position of inputs, the spatial context of the image can be lost.
+To address this, vision transformers often include positional encodings that capture the relative position of each patch within the image. By incorporating positional information, the model can better understand the spatial relationships between different patches and enhance its ability to recognize objects and patterns.
+
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx:12]:
```suggestion
```

File: chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+*__Note__: CNNs are designed to learn spatial features, while vision transformers are designed to learn both spatial and contextual features.*
+
+## Key Differences Between Image and Video Processing
+Videos are essentially a sequence of frames, and processing them requires techniques to capture and incorporate motion information. In image processing, the transformer ignores the temporal (time) relations between frames i.e. it only focuses on spatial (space) information of a frame.
```
Comment (merveenoyan) [inline chapters/en/Unit 7 - Video and Video Processing/video-processing-basics.mdx:16]:
```suggestion
Videos are essentially a sequence of frames, and processing them requires techniques to capture and incorporate motion information. In image processing, the transformer ignores the temporal (time) relations between frames, i.e., it only focuses on a frame's spatial (space) information.

```

**Review by merveenoyan — COMMENTED** [review]
Mostly readability nits, thank you!

**Review by merveenoyan — APPROVED** [review]
LGTM!

**Comment by DiwakarBasnet** [comment]
> LGTM!

Thanks @merveenoyan and @mmhamdy for the suggestions, do let me know if there are topics I need to add or focus more on

**Comment by ATaylorAerospace** [comment]
> > LGTM!
> 
> Thanks @merveenoyan and @mmhamdy for the suggestions, do let me know if there are topics I need to add or focus more on

@DiwakarBasnet This chapter is very comprehensive and detailed. Thanks for you and the teams work on this chapter. It look ready to merge basedon Merve's comment so I will go ahead and merge shortly!

**Review by ATaylorAerospace — APPROVED** [review]

**Pull request merged**